CommonApi/src/main/java/mezz/jei/api/registration/JeiRegistrationDelegate.java

```diff
@@ -0,0 +1,29 @@
+package mezz.jei.api.registration;
+
+import mezz.jei.api.IModPlugin;
+
+import java.util.function.Consumer;
+
+/**
+ * Advanced API to provide a stand-between when JEI is calling methods on the registered plugins.
+ * Only one registration delegate may exist. By default, it simply calls the method with no further
+ * logic.
+ */
+public interface JeiRegistrationDelegate {
+
+	static void set(JeiRegistrationDelegate delegate) {
+		JeiRegistrationDelegateHolder.current = delegate;
+	}
+
+	static JeiRegistrationDelegate get() {
+		return JeiRegistrationDelegateHolder.current;
+	}
+
+	void callOnPlugin(JeiRegistrationStep step, IModPlugin plugin, Consumer<IModPlugin> func);
+
+}
+
+// Can't have a private in an interface, including an inner class. So this will do.
+class JeiRegistrationDelegateHolder {
+	static JeiRegistrationDelegate current = (step, plugin, func) -> func.accept(plugin);
+}
```

Comment:
as a design rule, I do not store global variables like this in the API

I think this could be implemented a little more cleanly if we did something like `IModPlugin`, where JEI can discover  them and use the non-JEI one if found (and crash or pick the first one if there are multiple).

We could create two new types: 

1. `IJeiRuntimePlugin` which has
`void registerRuntime(IRuntimeRegistration registration)`
and deprecate that method in `IModPlugin`

2. `IJeiPluginLoaderPlugin` (lol?) which has
`void callOnPlugin(JeiRegistrationStep step, IModPlugin plugin, Consumer<IModPlugin> func)`